crates/ress/provider/src/lib.rs

```diff
@@ -162,7 +162,11 @@ where
         let witness_state_provider = self.provider.state_by_block_hash(ancestor_hash)?;
         let mut trie_input = TrieInput::default();
         for block in executed_ancestors.into_iter().rev() {
-            trie_input.append_cached_ref(block.trie.as_ref().unwrap(), &block.hashed_state);
+            if let Some(trie_updates) = block.trie.as_ref() {
+                trie_input.append_cached_ref(trie_updates, &block.hashed_state);
+            } else {
+                trace!(target: "reth::ress_provider", ancestor = ?block.recovered_block().num_hash(), "Skipping missing trie updates");
```

Comment:
it's incorrect to just ignore the missed trie updates, we should return an error in this case